### PR TITLE
Describe multiple modes of Compass installation

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Describe multiple modes of Compass installation
- Update Compass tutorial

### CODE REVIEW

File: docs/compass/04-01-installation.md [hunk 1]
```diff
@@ -29,40 +103,126 @@ Follow these steps to install Compass on a cluster:
...
+1. Check the Compass installation progress. To do so, download the script and check the progress of the installation:
+    ```bash
+    source <(curl -s "https://storage.googleapis.com/kyma-development-artifacts/compass/${INSTALLATION_OPTION}/is-installed.sh")
+    ```
 
-For local development, install Compass along with the minimal Kyma installation from the `master` branch. To do so, run this script:
-```bash
-./installation/cmd/run.sh
-```
+Once Compass is installed, Runtime Agent will be configured to fetch configuration from the Compass installation within the same cluster.
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:135]:
```suggestion
Once Compass is installed, Runtime Agent will be configured to fetch the configuration from the Compass installation within the same cluster.
```
I would specify what configuration, e.g. `cluster configuration`, `Runtime configuration`, etc.

File: docs/compass/04-01-installation.md [hunk 2]
```diff
@@ -29,40 +103,126 @@ Follow these steps to install Compass on a cluster:
...
 
-You can also specify Kyma version, such as 1.6 or newer:
-```bash
-./installation/cmd/run.sh {VERSION}
-```
+### Local Minikube installation
+
+To install Compass and Runtime components on a Minikube, follow these steps:
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:139]:
```suggestion
To install Compass and Runtime components on Minikube, follow these steps:
```

File: docs/compass/04-01-installation.md [hunk 3]
```diff
@@ -29,40 +103,126 @@ Follow these steps to install Compass on a cluster:
...
+
+1. Use Kyma CLI to provision Minikube cluster:
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:141]:
```suggestion
1. Use the Kyma CLI to provision a Minikube cluster:
```

File: docs/compass/04-01-installation.md [hunk 4]
```diff
@@ -29,40 +103,126 @@ Follow these steps to install Compass on a cluster:
...
+
+    ```bash
+    kyma provision minikube
+    ```
+1. Prepare Installation CR and overrides:
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:146]:
```suggestion
1. Prepare the Installation CR and overrides:
```

File: docs/compass/04-01-installation.md [hunk 5]
```diff
@@ -29,40 +103,126 @@ Follow these steps to install Compass on a cluster:
...
+      name: disable-legacy-connectivity-override
+      namespace: kyma-installer
+      labels:
+        installer: overrides
+        kyma-project.io/installation: ""
+    data:
+      global.disableLegacyConnectivity: "true"
+    EOF
+    ```
+1. In order to reduce memory and CPU usage, from the downloaded `installer-cr-cluster-runtime.yaml.tpl` file, comment out the components you don't want to use, such as `monitoring`, `tracing`, `logging`, `kiali`.
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:164]:
```suggestion
1. In order to reduce memory and CPU usage, from the downloaded `installer-cr-cluster-runtime.yaml.tpl` file, comment out the components you don't want to use, such as `monitoring`, `tracing`, `logging`, and `kiali`.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 6]
```diff
@@ -24,17 +24,17 @@ For simplicity reasons, use the available Order Service as the sample external A
...
 
 1. Open a separate tab in your browser and go to `https://compass.{CLUSTER_DOMAIN}`. It will navigate you to the Compass UI. From the drop-down list on the top navigation panel, select the tenant you want to work on. For the purpose of this tutorial, select the `default` tenant. In the **Runtimes** tab, there is already the default `kymaruntime` that you can work on to complete this tutorial. Make sure that your Runtime is assigned to the `DEFAULT` scenario.
 
-2. In the left navigation panel, navigate to the **Application** tab and click **Create application...** to register your Application in Compass. Choose **From scratch** from the drop-down list. For the purpose of this tutorial, name your Application `test-app`. By default, your Application is assigned to the `DEFAULT` scenario.
+2. In the left navigation panel, navigate to the **Application** tab and click **Create application...** to register your Application in Compass. Choose **From scratch** from the drop-down list. For the purpose of this tutorial, name your Application `test-app`. By default, your Application is assigned to the `DEFAULT` scenario. Click **Create** button.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:33]:
```suggestion
2. In the left navigation panel, navigate to the **Application** tab and click **Create application...** to register your Application in Compass. Choose **From scratch** from the drop-down list. For the purpose of this tutorial, name your Application `test-app`. By default, your Application is assigned to the `DEFAULT` scenario. Click the **Create** button.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 7]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
 
 ### Use your Application in the Kyma Console UI
 
-1. Go back to the Kyma Console UI. You can see that the `test-app` Application is registered in the **Applications** view. Select `test-app` and bind it to your Namespace by selecting the **Create Binding** button.
+1. Go back to the Kyma Console UI. You can see that the `test-app` Application is registered in the **Applications/Systems** view. Select `test-app` and bind it to your Namespace by selecting the **Create Binding** button.
 
-2. From the drop-down list in the top-right corner, select your Namespace and go to the **Catalog** view. You will see your services available under the **Services** tab. Provision the service instance by choosing your Package and clicking the **Add** button in the top-right corner of the page.
+2. From the drop-down list in the top-right corner, select your Namespace and go to the **Catalog** view. You will see your services available under the **Services** tab. Provision the service instance by choosing your Package as selected plan and clicking the **Add** button in the top-right corner of the page.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:50]:
```suggestion
2. From the drop-down list in the top-right corner, select your Namespace and go to the **Catalog** view. You will see your services available under the **Services** tab. Provision the service instance by choosing your Package as selected plan and clicking the **Add** button in the top-right corner of the page.
```
`as selected plan` -> what do you mean?

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 8]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
-4. Expose your lambda:
+4. Bind your lambda:
 
     a. In the left navigation panel, go to the **Lambdas** tab and click the `call-order-service` lambda.
 
-    b. In the **Settings & Code** section, click the **Select Function Trigger** button and expose your lambda via HTTPS.
+    b. Scroll down to the end of the Configuration view and bind your lambda to your instance by clicking the **Create Service Binding** button in the **Service Binding** section. Choose the ServiceInstance you want to bind your lambda to and click **Create Service Binding**.
 
-    c. Untick the **Enable authentication** option as there is no need to secure the connection for the purpose of this tutorial. Click **Add**.
+    c. Save the settings in the right top-right corner of the page.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:60]:
```suggestion
    c. Save the settings in the top right corner of the page.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 9]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
 
 3. Create a lambda function. In the **Overview** tab, click the **Deploy new resource** button and upload the file with the [lambda function](./assets/lambda.yaml).
 
-4. Expose your lambda:
+4. Bind your lambda:
 
     a. In the left navigation panel, go to the **Lambdas** tab and click the `call-order-service` lambda.
 
-    b. In the **Settings & Code** section, click the **Select Function Trigger** button and expose your lambda via HTTPS.
+    b. Scroll down to the end of the Configuration view and bind your lambda to your instance by clicking the **Create Service Binding** button in the **Service Binding** section. Choose the ServiceInstance you want to bind your lambda to and click **Create Service Binding**.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:58]: Are they still lambdas or are they Functions now? 
```suggestion
    b. Scroll down to the end of the Configuration view and bind your function to your instance by clicking the **Create Service Binding** button in the **Service Binding** section. Choose the ServiceInstance you want to bind your function to and click **Create Service Binding**.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 10]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
 
-    c. Untick the **Enable authentication** option as there is no need to secure the connection for the purpose of this tutorial. Click **Add**.
+    c. Save the settings in the right top-right corner of the page.
 
-    d. Scroll down to the end of your lambda view and bind your lambda to your instance by clicking the **Create Service Binding** button in the **Service Binding** section. Choose the ServiceInstance you want to bind your lambda to and click **Create Service Binding**.
+    d. Click the **Lambdas** tab and wait until the lambda status is completed and marked as `1/1`.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:62]: Are they still lambdas or are they Functions now?
```suggestion
    d. Click the **Functions** tab and wait until the function status is completed and marked as `1/1`.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 11]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
 
-    e. Save the settings in the right top-right corner of the page.
+5. Expose your lambda:
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:64]: Are they still lambdas or are they Functions now?
```suggestion
5. Expose your function: 
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 12]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
+    d. Click the **Lambdas** tab and wait until the lambda status is completed and marked as `1/1`.
 
-    e. Save the settings in the right top-right corner of the page.
+5. Expose your lambda: 
 
-    f. Click the **Lambdas** tab and wait until the lambda status is completed and marked as `1/1`.
+    a. In the left navigation panel, go to the **API Rules** tab and click the **Add API Rule** button.
 
-5. Test your lambda. Navigate to your lambda and go to the **Testing** tab. After you click the **Send** button, you can see the following output in the **Response** field:
+    b. Type `test-lambda` in both Name and Hostname fields. From the **Service** dropdown, select `call-order-service`. Keep the **Access strategies** as they are.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:68]: Are they still lambdas or are they Functions now?
```suggestion
    b. Type `test-function` in both the **Name** and **Hostname** fields. From the **Service** dropdown, select `call-order-service`. Keep **Access strategies** as they are.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 13]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
+
+    c. Click **Create** button at the top right corner of the page.
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:70]:
```suggestion
    c. Click the **Create** button in the top right corner of the page.
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 14]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
+
+6. Test your lambda. Open terminal app and run the following command:
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:72]: Are they still lambdas or are they Functions now?
```suggestion
6. Test your function. Open the terminal and run the following command:
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 15]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
+    ```bash
+    curl https://test-lambda.{clusterDomain}
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:74]:
```suggestion
    curl https://test-lambda.{CLUSTER_DOMAIN}
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 16]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
+    ```
+
+    You can see the following output:
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:77]:
```suggestion
    This returns the following output:
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 17]
```diff
@@ -74,12 +83,18 @@ For simplicity reasons, use the available Order Service as the sample external A
     }
     ```
 
-    You can test your lambda by performing the following actions in the **Payload** field:
+    You can test your lambda by performing the following actions:
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:86]: Are they still lambdas or are they Functions now?
```suggestion
    You can test your function by performing the following actions:
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 18]
```diff
@@ -74,12 +83,18 @@ For simplicity reasons, use the available Order Service as the sample external A
...
 
     - `{"action":"add"}` - adds the new order
     - `{"action":"list"}` - lists all orders (this is the default command executed after you click the **Send** button)
     - `{"action":"delete"}` - deletes all the orders
 
+    To perform an action, run the following command:
+
+    ```bash
+    curl --header "Content-Type: application/json" --request POST --data '{selectedAction}' https://test-lambda.{clusterDomain}
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:95]: Are they still lambdas or are they Functions now?
```suggestion
    curl --header "Content-Type: application/json" --request POST --data '{SELECTED_ACTION}' https://test-function.{CLUSTER_DOMAIN}
```
If they're still lambdas, please edit my suggestion to apply the rest of the changes.

File: docs/compass/04-01-installation.md [hunk 19]
```diff
@@ -1,15 +1,89 @@
...
+
+1. Select installation option for Compass and Kyma. ​There are three possible installation options:
+
+    | Installation option     	| Value to use with the installation command   	| Example value          	|
+    |-------------------------	|-------------------	|-------------------------	|
+    | From the Compass `master` branch 	| `master`          	| `master`                	|
+    | From the specific commit on the Compass `master` branch 	| `master-{COMMIT_HASH}` 	| `master-34edf09a` 	|
+    | From the specific PR on the Compass repository       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
+
+    The Kyma version is read from the [`KYMA_VERSION`](../../installation/resources/KYMA_VERSION) file on a specific commit.
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply docs/compass/04-01-installation.md:21]: @majakurcius, no, the specific commit is checked out and then the KYMA_VERSION is read. So the file is read from a specific commit in history.

File: docs/compass/04-01-installation.md [hunk 20]
```diff
@@ -1,15 +1,89 @@
...
+
+    Once you decide on the installation option, use this command:
+
+    ```bash
+    export INSTALLATION_OPTION={installationOption}
+    ```
+
+1. Prepare the cluster for custom installation. Read how to prepare the cluster [with `xip.io` domain](https://kyma-project.io/docs/#installation-install-kyma-on-a-cluster-prepare-the-cluster) or [with custom domain](https://kyma-project.io/docs/#installation-install-kyma-with-your-own-domain-prepare-the-cluster). Remember to apply all global overrides twice - in both `kyma-installer` and `compass-installer` namespaces.
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply docs/compass/04-01-installation.md:29]: 😄 Okay, fixed!

File: docs/compass/08-01-basic-flow-tutorial.md (hunk 7 above)
Comment (pkosiec) [reply docs/compass/08-01-basic-flow-tutorial.md:50]: There is a dropdown called "Selected plan". From here you have to select your Package.

File: docs/compass/08-01-basic-flow-tutorial.md (hunk 10 above)
Comment (pkosiec) [reply docs/compass/08-01-basic-flow-tutorial.md:62]: Right, they're Functions now.

File: docs/compass/04-01-installation.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/compass/04-01-installation.md:21]: Oh, okay, gotcha.

File: docs/compass/04-01-installation.md [hunk 21]
```diff
@@ -1,15 +1,89 @@
...
+1. Apply overrides using the following command from the root directory of the Compass repository:
+
+    ```bash
+    kubectl create namespace kyma-installer || true \
+        && kubectl apply -f ./installation/resources/installer-overrides-compass-gateway.yaml
+    ```
+
+2. Perform minimal Kyma installation with the following command:
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply docs/compass/04-01-installation.md:37]: It should be a single list here. Something's wrong with ordering, so I will fix it soon 🙂

File: docs/compass/08-01-basic-flow-tutorial.md (hunk 7 above)
Comment (majakurcius) [reply docs/compass/08-01-basic-flow-tutorial.md:50]:
```suggestion
2. From the drop-down list in the top-right corner, select your Namespace and go to the **Catalog** view. You will see your services available under the **Services** tab. Provision the service instance by choosing your Package as **Selected plan** from the dropdown list and clicking the **Add** button in the top-right corner of the page.
```
Or you can skip the `from the dropdown list` if you think it's too lengthy.

File: docs/compass/04-01-installation.md [hunk 22]
```diff
@@ -29,40 +103,126 @@ Follow these steps to install Compass on a cluster:
...
+    kubectl get configmap -n kyma-installer {override-name} -oyaml --export | kubectl apply -n compass-installer -f -
+    ```
+
+1. Install Compass. ​There are three possible installation options:
 
     | Installation option     	| Value to use with the installation command   	| Example value          	|
     |-------------------------	|-------------------	|-------------------------	|
     | From the `master` branch 	| `master`          	| `master`                	|
     | From the specific commit on the `master` branch 	| `master-{COMMIT_HASH}` 	| `master-34edf09a` 	|
     | From the specific PR       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply docs/compass/04-01-installation.md:118]: Okay, as discussed we will keep the tables duplicated to enhance readability of the document.

File: docs/compass/04-01-installation.md [hunk 23]
```diff
@@ -1,15 +1,86 @@
...
+    | From a specific PR on the Compass repository       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
+
+    The Kyma version is read from the [`KYMA_VERSION`](../../installation/resources/KYMA_VERSION) file on a specific commit.
+
+    Once you decide on the installation option, use this command:
+
+    ```bash
+    export INSTALLATION_OPTION={CHOSEN_INSTALLATION_OPTION_HERE}
+    ```
+1. Prepare the cluster for custom installation. Read how to prepare the cluster [with the `xip.io` domain](https://kyma-project.io/docs/#installation-install-kyma-on-a-cluster-prepare-the-cluster) or [with a custom domain](https://kyma-project.io/docs/#installation-install-kyma-with-your-own-domain-prepare-the-cluster). Remember to apply all global overrides both in the `kyma-installer` and `compass-installer` Namespaces.
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:28]:
```suggestion
1. Prepare the cluster for custom installation. Read how to prepare the cluster [with the `xip.io` domain](https://kyma-project.io/docs/#installation-install-kyma-on-a-cluster-prepare-the-cluster) or [with a custom domain](https://kyma-project.io/docs/#installation-install-kyma-with-your-own-domain-prepare-the-cluster). Remember to apply all global overrides in both the `kyma-installer` and `compass-installer` Namespaces.
```

File: docs/compass/04-01-installation.md [hunk 24]
```diff
@@ -29,40 +100,127 @@ Follow these steps to install Compass on a cluster:
...
+
+    ```bash
+    kubectl get configmap -n kyma-installer {OVERRIDE_NAME} -oyaml --export | kubectl apply -n compass-installer -f -
+    ```
+1. Install Compass. ​There are three possible installation options:
 
     | Installation option     	| Value to use with the installation command   	| Example value          	|
     |-------------------------	|-------------------	|-------------------------	|
     | From the `master` branch 	| `master`          	| `master`                	|
     | From the specific commit on the `master` branch 	| `master-{COMMIT_HASH}` 	| `master-34edf09a` 	|
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:117]:
```suggestion
    | From a specific commit on the `master` branch 	| `master-{COMMIT_HASH}` 	| `master-34edf09a` 	|
```

File: docs/compass/04-01-installation.md [hunk 25]
```diff
@@ -29,40 +100,127 @@ Follow these steps to install Compass on a cluster:
...
     | From the specific PR       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:118]:
```suggestion
    | From a specific PR       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
```

File: docs/compass/04-01-installation.md [hunk 26]
```diff
@@ -29,40 +100,127 @@ Follow these steps to install Compass on a cluster:
...
 
-    Once you decide on the installation option, use this command:
+    Once you decide on the installation option, use these commands:
+    
     ```bash
-    kubectl apply -f https://storage.cloud.google.com/kyma-development-artifacts/compass/{INSTALLATION_OPTION}/compass-installer.yaml
+    export INSTALLATION_OPTION={installationOption}
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:123]:
```suggestion
    export INSTALLATION_OPTION={CHOSE_INSTALLATION_OPTION}
```

File: docs/compass/04-01-installation.md [hunk 27]
```diff
@@ -29,40 +100,127 @@ Follow these steps to install Compass on a cluster:
...
+    $(sed -e 's/\.minikubeIP: .*/\.minikubeIP: '"${MINIKUBE_IP}"'/g' installation/resources/installer-config-local.yaml.tpl)
+    EOF
+    ```
+
+1. Install Compass. ​There are three possible installation options:
+    
+    | Installation option     	| Value to use with the installation command   	| Example value          	|
+    |-------------------------	|-------------------	|-------------------------	|
+    | From the `master` branch 	| `master`          	| `master`                	|
+    | From the specific commit on the `master` branch 	| `master-{COMMIT_HASH}` 	| `master-34edf09a` 	|
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:206]:
```suggestion
    | From a specific commit on the `master` branch 	| `master-{COMMIT_HASH}` 	| `master-34edf09a` 	|
```

File: docs/compass/04-01-installation.md [hunk 28]
```diff
@@ -29,40 +100,127 @@ Follow these steps to install Compass on a cluster:
...
+    | From the specific PR       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
```
Comment (majakurcius) [inline docs/compass/04-01-installation.md:207]:
```suggestion
    | From a specific PR       	| `PR-{PR_NUMBER}`         	| `PR-1420`     	|
```

File: docs/compass/08-01-basic-flow-tutorial.md [hunk 29]
```diff
@@ -45,27 +45,36 @@ For simplicity reasons, use the available Order Service as the sample external A
...
+    a. In the left navigation panel, go to the **API Rules** tab and click the **Add API Rule** button.
 
-5. Test your lambda. Navigate to your lambda and go to the **Testing** tab. After you click the **Send** button, you can see the following output in the **Response** field:
+    b. Type `test-function` in both the **Name** and **Hostname** fields. From the **Service** dropdown, select `call-order-service`. Keep **Access strategies** as they are.
+
+    c. Click the **Create** button in the top right corner of the page.
+
+6. Test your function. Open the terminal and run the following command:
+    ```bash
+    curl https://test-lambda.{CLUSTER_DOMAIN}
```
Comment (majakurcius) [inline docs/compass/08-01-basic-flow-tutorial.md:74]:
```suggestion
    curl https://test-function.{CLUSTER_DOMAIN}
```
I guess this one should be `test-function`, too, right?

**Pull request merged**